SULLesson10_12/SULLesson10_12/AuthorizationScreen/AuthorizationScreen.swift

```diff
@@ -0,0 +1,191 @@
+//
+//  AuthorizationScreen.swift
+//  SULLesson10_12
+//
+//  Created by Григоренко Александр Игоревич on 24.01.2023.
+//
+
+import SwiftUI
+
+/// Экран авторизации
+struct AuthorizationView: View {
+
+    // MARK: - Public enums
+
+    enum Scenario: String {
+        case LogIn = "Log in"
+        case SingUp = "Sing up"
+    }
+
+    // MARK: - Private constants
+
+    private enum Constants {
+        static let mask = "+0(000) 000-00-00"
+        static let backgroundViewColorOpacity: Double = 1
+        static let emptyString = ""
+        static let authorizationViewYOffset: CGFloat = 30
+        static let screenScenarioPickerViewCornerRadius: CGFloat = 15
+        static let screenScenarioPickerViewWidth: CGFloat = 150
+        static let screenScenarioPickerViewScaleCoef: CGFloat = 2
+        static let screenScenarioPickerViewScaleEdgeInsets = EdgeInsets(top: 50, leading: 0, bottom: 50, trailing: 0)
+        static let loginButtonText = "Log in"
+        static let signUpButtonText = "Sign up"
+        static let passwordTitleText = "Password"
+        static let checkVerificationButtonText = "Check verification"
+        static let secureFieldPlaceholder = "● ● ● ● ● ● ●"
+        static let forgotYourPasswordLinkText = "Forgot your password?"
+        static let forgotYourPasswordAlertText = "Forgot your password? Call 8-800-555-35-35"
+        static let maxCountOfLoginText = 17
+    }
+
+    // MARK: - public properties
+
+    var body: some View {
+        backgroundView
+    }
+
+    // MARK: - init
+
+    init() {
+        UITextField.appearance().textColor = .black
+    }
+
+    // MARK: - private properties
+
+    @FocusState private var loginFocusState: Bool
+    @FocusState private var passwordFocusState: Bool
+    @ObservedObject private var viewModel = AuthorizationViewModel()
+    @State private var scenarioPickerSelect = Scenario.LogIn
+    @State private var loginText = Constants.emptyString
+    @State private var passwordText = Constants.emptyString
+    @State private var logInSegmentButton = Color.white
+    @State private var signUpSegmentButton = Color.gray
+    @State private var isForgotPasswordShown = false
```

Comment:
```suggestion
    @State private var scenarioPickerSelect = Scenario.LogIn
    @State private var loginText = Constants.emptyString
    @State private var passwordText = Constants.emptyString
    @State private var logInSegmentButton = Color.white
    @State private var signUpSegmentButton = Color.gray
    @State private var isForgotPasswordShown = false
```

это надо было во вьюмодель вынести